exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -220,7 +225,21 @@ public QProfiles getProfilesJSON() {
 
       final List<ProfileInfo> finishedQueries = Lists.newArrayList();
 
-      final Iterator<Map.Entry<String, QueryProfile>> range = completed.getRange(0, MAX_PROFILES);
+      //Defining #Profiles to load
+      int maxProfilesToLoad;
+      if (MaxProfiles == null) {
+        MaxProfiles = work.getContext().getConfig().getInt(ExecConstants.HTTP_MAX_PROFILES);
```

Comment:
I wasn't sure if I'd want to repeatedly load the value from the config object, as I was assuming it might have synchronized access. But considering this is not going to be called at a very high frequency, there shouldn't be any overhead.